apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceHeaderContext.cs

```diff
@@ -28,6 +28,8 @@ namespace Google.Cloud.Diagnostics.Common
     /// </remarks>
     internal sealed class TraceHeaderContext
     {
+        private static readonly TraceIdFactory _traceIdFactory = TraceIdFactory.Create();
```

Comment:
None really, I think the TraceIdFactory should be a singleton. 

I don't quite follow the second part of your comment, are you referring to "FromHeader"?